### PR TITLE
Added JS to add category as a class for additional styling purposes

### PR DESCRIPTION
**Pull request by lagwagon**
I thought that category-based (by name) styling could be useful (specifically for my use, possibly others?). Maybe not useful for the average end user, but for those of us who wish to theme our own websites it could be useful. 

Added this in and it works well. 

categories

### CODE REVIEW

**Comment by coding-horror** [comment]
shouldn't it be `category-foo` ? so that things sort properly?

File: app/assets/javascripts/discourse/views/topic-list-item.js.es6 [hunk 1]
```diff
@@ -10,8 +10,17 @@ export default Discourse.View.extend(StringBuffer, {
...
   unboundClassNames: function(){
-    var classes = [];
     var topic = this.get('topic');
+    var classes = [];
+    
+    if(topic.get('category')){
+      var category, category_name, category_class;
+      category = topic.get('category');
+      category_name = category.get('name').toLowerCase();
+      category_class = category_name + '-category';
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/views/topic-list-item.js.es6]: This should be replaced by this oneliner

```javascript
var className = "category-" + topic.get('category.nameLower');
```

Comment (SamSaffron) [inline app/assets/javascripts/discourse/views/topic-list-item.js.es6]: what if a category name has a space in it or something that needs url
encoding? I am uncertain about this change, feels like custom theme
territory.

Comment (ZogStriP) [inline app/assets/javascripts/discourse/views/topic-list-item.js.es6]: Yeah, should probably use the slug instead

**Comment by lagwagon** [comment]
Hey guys - Thanks for the comments. I've made some changes, namely to use the slug, and used Jeff's suggestion for how the class name appears as it does look better. 

The slug already takes care of escaping html characters and it seems to be fairly reliable in providing something that meshes well with the CSS overall.

Right now there is no way to style posts made in certain categories in a friendly way. This is a pretty huge thing that is found in nearly every cms or board software out there, I figured it's a nice-to-have.

File: app/assets/javascripts/discourse/views/topic-list-item.js.es6 [hunk 2]
```diff
@@ -10,8 +10,17 @@ export default Discourse.View.extend(StringBuffer, {
...
     var topic = this.get('topic');
+    var classes = [];
+
+    if(topic.get('category')){
+      var category,categorySlug,categoryClass;
+      category = topic.get('category');      
+      categorySlug = category.get('slug');
+      categoryClass = 'category-' + categorySlug;
+
+      classes.push(categoryClass);
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/views/topic-list-item.js.es6]: `classes.push("category-" + topic.get('category.slug'))` ;)

Comment (lagwagon) [inline app/assets/javascripts/discourse/views/topic-list-item.js.es6]: Haha okay, made that change too. Thanks for the advice! I generally am not writing javascript...ever.. anymore, so every comment helps me

**Pull request closed without merging**